### PR TITLE
feat(pt): add universal test for loss

### PR DESCRIPTION
**Pull request by iProzd**
## Release Notes

- **New Features**
	- Introduced a new `LossTest` class for enhanced testing of loss functions.
	- Added multiple parameterized test functions for various loss functions in the new `test_loss.py` file.
  
- **Bug Fixes**
	- Corrected tensor operations in the `DOSLoss` class to ensure accurate cumulative sum calculations.

- **Documentation**
	- Added SPDX license identifiers to multiple files for clarity on licensing terms.

- **Chores**
	- Refactored data conversion methods in the `PTTestCase` class for improved handling of tensors and arrays.

### CODE REVIEW

File: source/tests/universal/common/cases/loss/utils.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        natoms = 5
+
+        def fake_model():
+            model_predict = {
+                data_key: fake_input_one_frame(label_dict[data_key], natoms)
+                for data_key in label_keys
+            }
+            if "atom_ener" in model_predict:
+                model_predict["atom_energy"] = model_predict.pop("atom_ener")
+            model_predict.update({"mask_mag": np.ones([1, natoms, 1], dtype=np.bool_)})
```
Comment (wanghan-iapcm) [inline source/tests/universal/common/cases/loss/utils.py:44]: we are using int32 for masks. 
 https://github.com/deepmodeling/deepmd-kit/blob/bb6b06b12955d3bf2b126e11e54f30fe2dd8bbb2/deepmd/dpmodel/atomic_model/base_atomic_model.py#L204

Comment (njzjz) [reply source/tests/universal/common/cases/loss/utils.py:44]: Indeed I feel `bool` is more reasonable for 0-1 values. I have forgotten the reason why int32 is used...

File: source/tests/universal/pt/loss/test_loss.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+@parameterized(
+    (
+        *[(param_func, EnergyStdLoss) for param_func in LossParamEnergyList],
+        *[(param_func, EnergySpinLoss) for param_func in LossParamEnergySpinList],
+        *[(param_func, DOSLoss) for param_func in LossParamDosList],
+        *[(param_func, TensorLoss) for param_func in LossParamTensorList],
+        *[(param_func, PropertyLoss) for param_func in LossParamPropertyList],
+    )  # class_param & class
+)
+class TestFittingPT(unittest.TestCase, LossTest, PTTestCase):
```
Comment (njzjz) [inline source/tests/universal/pt/loss/test_loss.py:39]:
```suggestion
class TestLossPT(unittest.TestCase, LossTest, PTTestCase):
```

File: source/tests/universal/common/cases/loss/utils.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            return model_predict
+
+        labels = {
+            data_key: fake_input_one_frame(label_dict[data_key], natoms)
```
Comment (njzjz) [inline source/tests/universal/common/cases/loss/utils.py:48]: Testing at least two frames is better.

File: source/tests/universal/common/cases/loss/utils.py (hunk 1 above)
Comment (wanghan-iapcm) [reply source/tests/universal/common/cases/loss/utils.py:44]: I agree. We may need a PR to clean up.

Comment (iProzd) [reply source/tests/universal/common/cases/loss/utils.py]: The output mask `mask_mag` for virtual atoms in spin model is `bool`:
https://github.com/deepmodeling/deepmd-kit/blob/bb6b06b12955d3bf2b126e11e54f30fe2dd8bbb2/deepmd/pt/model/model/spin_model.py#L128

Maybe we keep this `bool` and clean up other int32 mask in another PR ?

**Review by njzjz — APPROVED** [review]

File: source/tests/universal/common/cases/loss/utils.py (hunk 1 above)
Comment (njzjz) [reply source/tests/universal/common/cases/loss/utils.py]: I agree

**Pull request merged**